### PR TITLE
MHR Owner Role and Additional Name Updates

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#16176

*Description of changes:*
* Major refactor on Home Owner roles to include sub component and Owner Role Configurations
* Minor configuration updates to Additional Name Field

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1322-8xlse9xd.web.app

**Comment by cameron-eyds** [comment]
Hide white space for easier reviewing!

Also, just working on some additional unit testing. Will commit shortly 👍

File: ppr-ui/src/components/mhrRegistration/HomeOwners/AddEditHomeOwner.vue [hunk 1]
```diff
@@ -9,74 +9,8 @@
...
-                  isTransferToExecutorUnder25Will ||
-                  isTransferToExecutorProbateWill ||
-                  isFrozenMhr"
-                v-model="HomeOwnerPartyTypes.ADMINISTRATOR"
-              />
-            </v-radio-group>
-          </v-col>
-        </v-row>
+        <!-- Owner Roles Component-->
+        <HomeOwnerRoles :partyType="owner.partyType"  @update:partyType="owner.partyType = $event"/>
```
Comment (dimak1) [inline ppr-ui/src/components/mhrRegistration/HomeOwners/AddEditHomeOwner.vue:13]: What if we update the `owner.partyType` in the child component without emitting? It is still reactive, right. Or you wanted to follow the pattern of emitting events up?

File: ppr-ui/src/components/mhrRegistration/HomeOwners/AddEditHomeOwner.vue [hunk 2]
```diff
@@ -678,6 +608,22 @@ export default defineComponent({
         return !!localState.groupFractionalData?.interestNumerator &&
           !!localState.groupFractionalData?.interestDenominator
       }),
+      nameConfig: computed((): AdditionalNameConfigIF => {
+        return AdditionalNameConfig[localState.owner.partyType]
+      }),
+      additionalNameRules: computed((): Array<Function> => {
+        return localState.nameConfig.isRequired
```
Comment (dimak1) [inline ppr-ui/src/components/mhrRegistration/HomeOwners/AddEditHomeOwner.vue:615]: 👍🏻 nice implementation of required field

**Review by dimak1 — APPROVED** [review]
Looks great! Very nice refactor and implementation of the roles config. 🥇

**Comment by dimak1** [comment]
@cameron-eyds we should merge this PR before mine. I did a small refactor and there might be conflicts that I might need to resolve.

File: ppr-ui/src/components/mhrRegistration/HomeOwners/AddEditHomeOwner.vue (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/mhrRegistration/HomeOwners/AddEditHomeOwner.vue:13]: You bet, it's reactive both ways. 
I don't believe you could update it in the child without it emitting. Is there a specific scenario you are thinking of?

**Comment by cameron-eyds** [comment]
> @cameron-eyds we should merge this PR before mine. I did a small refactor and there might be conflicts that I might need to resolve.

Sounds good!

**Review by doug-lovett — APPROVED** [review]
Looks spot on.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1322-8xlse9xd.web.app

**Pull request merged**